src/ddmd/inline.d

```diff
@@ -1524,14 +1533,39 @@ bool canInline(FuncDeclaration fd, bool hasthis, bool hdrscan, bool statementsTo
         if (tf.varargs == 1)
             goto Lno;
 
+        static bool isArrayOperationFunction(FuncDeclaration fd)
+        {
+            auto p = fd.ident.toChars();
+            if (memcmp(p, "_array".ptr, 6) == 0) // better to set a bit in the FuncDeclaration than this
```

Comment:
That's the one!

We really need to fix the hacks surrounding array ops' evaluation ordering.  They absolutely should be inlined when it makes sense.